### PR TITLE
[libSyntax] Record reused node IDs

### PR DESCRIPTION
**Pull request by ahoppen**
This is cheaper than recording reused region offsets and the reused node IDs will later be used to incrementally transfer the syntax to SwiftSyntax.

### CODE REVIEW

**Comment by ahoppen** [comment]
@swift-ci Please smoke test

File: lib/Parse/SyntaxParsingCache.cpp [hunk 1]
```diff
@@ -84,10 +85,52 @@ llvm::Optional<Syntax> SyntaxParsingCache::lookUp(size_t NewPosition,
...
+    bool didReuseNode(unsigned NodeId) {
+      return ReusedNodeIds.count(NodeId) > 0;
+    }
+
+  public:
+    ReusedRegionsCollector(std::unordered_set<unsigned> ReusedNodeIds)
+        : ReusedNodeIds(ReusedNodeIds) {}
+
+    const std::vector<SyntaxReuseRegion> &getReusedRegions() {
+      std::sort(ReusedRegions.begin(), ReusedRegions.end(), [](const SyntaxReuseRegion &Lhs, const SyntaxReuseRegion &Rhs) -> bool {
```
Comment (nkcsgexi) [inline lib/Parse/SyntaxParsingCache.cpp]: Nit: 80 column; otherwise LGTM!

**Review by nkcsgexi — APPROVED** [review]

File: include/swift/Parse/SyntaxParsingCache.h [hunk 2]
```diff
@@ -86,16 +80,15 @@ class SyntaxParsingCache {
...
 
-  /// Turn recording of reused ranges on
-  void setRecordReuseInformation() { RecordReuseInformation = true; }
-
-  /// Return the ranges of the new source file that have been successfully
-  /// looked up in this cache as a (start, end) pair of byte offsets in the
-  /// post-edit file.
-  std::vector<SyntaxReuseRegion> getReusedRanges() const {
-    return ReusedRanges;
+  const std::unordered_set<unsigned> &getReusedNodeIds() const {
```
Comment (nkcsgexi) [inline include/swift/Parse/SyntaxParsingCache.h]: It would be good to have a type alias of unsigned as the ID type, so we can search more easily.

**Comment by ahoppen** [comment]
@nkcsgexi Fixed the 80-column issue and added a typedef for `SyntaxNodeId`s.

@swift-ci Please smoke test

**Pull request merged**